feature/home/src/main/java/io/filmtime/feature/home/HomeViewModel.kt

```diff
@@ -27,53 +28,81 @@ class HomeViewModel @Inject constructor(
   val state = _state.asStateFlow()
 
   init {
-    viewModelScope.launch {
-      loadTrendingMovies()
-      loadTrendingShows()
-    }
+
+    loadTrendingMovies()
+    loadTrendingShows()
+
   }
 
-  private suspend fun loadTrendingMovies() {
-    getTrendingMovies()
-      .onStart {
-        _state.update { state -> state.copy(isLoading = true) }
-      }
-      .onCompletion { _state.update { state -> state.copy(isLoading = false) } }
-      .onEach { result ->
-        when (result) {
-          is Success -> {
-            _state.update { state ->
-              state.copy(videoSections = state.videoSections + listOf(VideoSection("Trending Movies", result.data)))
+  fun loadTrendingMovies() {
+    viewModelScope.launch {
+      getTrendingMovies()
+        .onStart {
+          _state.update { state -> state.copy(isLoading = true) }
+
```

Comment:
as I mentioned before, you we can call `loadTrendingMovies` and `loadTrendingShows` in `retry` function. so there is no need to make this functions public